src/backend/distributed/transaction/backend_data.c

```diff
@@ -780,6 +805,113 @@ MarkCitusInitiatedCoordinatorBackend(void)
 }
 
 
+/*
+ * AssignGlobalPID assigns a global process id for the current backend if
+ * it is not already assigned
```

Comment:
might use some comment. When do we generate new one? When do we use the globalPID assigned to this process?